File: _ai_contest_/clientPack/DAD_AI_2017_pack/BotJava/AI.java
public class AI {
	private static AI instance = null;
	private int[][] m_board = new int[Client.MAP_SIZE][Client.MAP_SIZE];
	private int m_index;
	PlayerInfos m_myInfos;
	PlayerInfos m_enemyInfos;

	public static AI getInstance() {
		if (instance == null) {
			instance = new AI();
		}
		return instance;
	}

	public int[][] getBoard() {
		return this.m_board;
	}

	void setBoard(int[][] board) {
		for (int i = 0; i < Client.MAP_SIZE; i++) {
			for (int j = 0; j < Client.MAP_SIZE; j++) {
				this.m_board[i][j] = board[i][j];
			}
		}
	}

	public int getPlayerIndex() {
		return this.m_index;
	}

	void setPlayerIndex(int paramInt) {
		this.m_index = paramInt;
	}

	public int getBlock(Position paramPosition) {
		if ((paramPosition.x < 0) || (paramPosition.x >= Client.MAP_SIZE) || (paramPosition.y < 0) || (paramPosition.y >= Client.MAP_SIZE)) {
			return -1;
		}
		return this.m_board[paramPosition.x][paramPosition.y];
	}

	public PlayerInfos getMyInfos() {
		return this.m_myInfos;
	}

	void setMyInfos(PlayerInfos myInfos) {
		this.m_myInfos = myInfos;
	}

	public PlayerInfos getEnemyInfos() {
		return this.m_enemyInfos;
	}

	void setEnemyInfos(PlayerInfos enemyInfos) {
		this.m_enemyInfos = enemyInfos;
	}
}

File: _ai_contest_/clientPack/DAD_AI_2017_pack/Bot C++/AI/src/AI.cpp
#include "AI.h"
#include <windows.h>

using namespace std;

AI* AI::s_instance = NULL;

AI::AI() : Update (NULL), m_myPosition(Position(0, 0)), m_enemyPosition(Position(0, 0))
{}

int * AI::GetBoard()
{
	return m_board;
}

void AI::SetBoard(int * board)
{
	for(int i = 0; i < MAP_SIZE * MAP_SIZE; i++)
	{
		m_board[i] = board[i];
	}
}

int AI::GetPlayerIndex()
{
	return m_index;
}

void AI::SetPlayerIndex(int idx)
{
	m_index = idx;
}

int AI::GetBlock(Position pos)
{
	if(pos.x < 0 || pos.x >= MAP_SIZE || pos.y < 0 || pos.y >= MAP_SIZE)
	{
		return BLOCK_OUT_OF_BOARD;
	}
	else
	{
		int idx = CONVERT_COORD(pos.x, pos.y);
		return m_board[idx];
	}
}

Position AI::GetMyPosition()
{
	return m_myPosition;
}

void AI::SetMyPosition(Position newPos)
{
	m_myPosition = newPos;
}

Position AI::GetEnemyPosition()
{
	return m_enemyPosition;
}

void AI::SetEnemyPosition(Position newPos)
{
	m_enemyPosition = newPos;
}

void AI::SetListBallMap(std::vector<Position> list){
	m_listBall = list;
}
void AI::SetListHardBlock(std::vector<Position> list){
	m_listHardBlock = list;
}
void AI::SetListSoftBlock(std::vector<Position> list){
	m_listSoftBlock = list;
}

void AI::SetMyBlockStack(int numBlock){
	m_myBlockInStack = numBlock;
}
void AI::SetEnemyBlockStack(int numBlock){
	m_enemyBlockInStack = numBlock;
}
void AI::SetMyPointScore(int score){
	m_myPointScore = score;
}
void AI::SetEnemyPointScore(int score){
	m_enemyPointScore = score;
}

std::vector<Position> AI::GetListBall(){
	return m_listBall;
}
std::vector<Position> AI::GetListHardBlock(){
	return m_listHardBlock;
}
std::vector<Position> AI::GetListSoftBlock(){
	return m_listSoftBlock;
}

int AI::GetMyBlockStack(){
	return m_myBlockInStack;
}
int AI::GetEnemyBlockStack(){
	return m_enemyBlockInStack;
}
int AI::GetMyPoint(){
	return m_myPointScore;
}
int AI::GetEnemyPoint(){
	return m_enemyPointScore;
}
